metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/MemoryAggregator.java

```diff
@@ -0,0 +1,70 @@
+package io.dockstore.metricsaggregator.helper;
+
+import io.dockstore.metricsaggregator.DoubleStatistics;
+import io.dockstore.openapi.client.model.MemoryMetric;
+import io.dockstore.openapi.client.model.Metrics;
+import io.dockstore.openapi.client.model.RunExecution;
+import io.dockstore.openapi.client.model.TaskExecutions;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Aggregate Memory metrics by calculating the minimum, maximum, and average.
+ */
+public class MemoryAggregator implements RunExecutionAggregator<MemoryMetric, Double> {
+    @Override
+    public MemoryMetric getMetricFromMetrics(Metrics metrics) {
+        return metrics.getMemory();
+    }
+
+    @Override
+    public Double getMetricFromRunExecution(RunExecution runExecution) {
+        return runExecution.getMemoryRequirementsGB();
+    }
+
+    @Override
+    public Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun) {
```

Comment:
This is the new function that aggregates tasks into a workflow execution for the memory metric